src/variable.jl

```diff
@@ -309,85 +309,73 @@ end
-            return data
-        end
+_read_data_from_nc!(v::Variable, aout) = _read_data_from_nc!(v, aout, 1)
 
-        Base.setindex!(v::Variable,data::$data_type,indexes::Colon...) = setindex!(v::Variable,data)
+function writeblock!(v::Variable, data, indexes::TI...) where TI <: Union{AbstractUnitRange,StepRange}
+    datamode(v.ds)
+    _write_data_to_nc(v, data, indexes...)
+    return data
+end
 
-        function Base.setindex!(v::Variable{T,N},data::$data_type,indexes::StepRange{Int,Int}...) where {T,N}
-            datamode(v.ds) # make sure that the file is in data mode
-            start,count,stride,jlshape = ncsub(indexes[1:ndims(v)])
-            tmp = fill(convert(T,data),jlshape)
-            nc_put_vars(v.ds.ncid,v.varid,start,count,stride,tmp)
-            return data
-        end
-    end
+function _write_data_to_nc(v::Variable{T,N},data,indexes::Int...) where {T,N}
+    nc_put_var1(v.ds.ncid,v.varid,[i-1 for i in reverse(indexes)],T(data[1]))
 end
 
-function Base.setindex!(v::Variable{T,N},data::AbstractArray{T,N},indexes::Colon...) where {T,N}
-    datamode(v.ds) # make sure that the file is in data mode
+_write_data_to_nc(v::Variable, data) = _write_data_to_nc(v, data, 1)
 
-    nc_put_var(v.ds.ncid,v.varid,data)
-    return data
+function _write_data_to_nc(v::Variable{T, N}, data, indexes::StepRange{Int,Int}...) where {T, N}
+    start,count,stride,jlshape = ncsub(indexes)
+    nc_put_vars(v.ds.ncid,v.varid,start,count,stride,T.(data))
 end
 
-function Base.setindex!(v::Variable{T,N},data::AbstractArray{T2,N},indexes::Colon...) where {T,T2,N}
-    datamode(v.ds) # make sure that the file is in data mode
-    tmp =
-        if T <: Integer
-            round.(T,data)
-        else
-            convert(Array{T,N},data)
-        end
+function _write_data_to_nc(v::Variable, data, indexes::Union{AbstractRange{<:Integer}}...)
+    ind = prod(length.(indexes)) == 1 ? first.(indexes) : normalizeindexes(size(v),indexes)
+    return _write_data_to_nc(v, data, ind...)
+end
 
-    nc_put_var(v.ds.ncid,v.varid,tmp)
-    return data
+function grow!(v::CFVariable, data, indexes::Union{Integer, Colon}...)
+    unlimdims = unlimited(Dimensions(CommonDataModel.dataset(v)))
+    length(unlimdims) == 1 || error("Only 1 unlimited dimension is supported") 
+    alldims = dimnames(v)
+    iunlimdim = findfirst(==(unlimdims[1]), alldims)
+    icol = findfirst(x -> x isa Colon, indexes)
+    iunlimdim == icol || ArgumentError("The Colon in the arguments must match the unlimited dimension.")
+    inds = replace(indexes, Colon() => 1:length(data))
+    v[inds...] = data
+    return v
 end
 
+getchunksize(v::Variable) = getchunksize(haschunks(v),v)
+getchunksize(::DiskArrays.Chunked, v::Variable) = chunking(v)[2]
+# getchunksize(::DiskArrays.Unchunked, v::Variable) = DiskArrays.estimate_chunksize(v)
+getchunksize(::DiskArrays.Unchunked, v::Variable) = size(v)
+eachchunk(v::CFVariable) = eachchunk(v.var)
```

Comment:
These are defined on `CFVariable`, but it's not implementing the rest of the DiskArrays.jl interface